test/pre-commit-hook.integration-test.js

```diff
@@ -0,0 +1,181 @@
+'use strict';
+
+const shell = require('shelljs');
+const fs = require('fs');
+
+const mockers = require('./test-utils/mockers');
+
+// Jest swallows most standard console logs not explicitly defined into a custom logger
+// see: https://stackoverflow.com/questions/51555568/remove-logging-the-origin-line-in-jest
+const consoleWarnSpy = jest.spyOn(global.console, 'warn').mockImplementation();
+
+const consoleInfoSpy = jest.spyOn(global.console, 'info').mockImplementation();
+
+const consoleErrorSpy = jest
+  .spyOn(global.console, 'error')
+  .mockImplementation();
+
+function exec(opt = '') {
+  if (typeof opt === 'string') {
+    const cli = require('../command');
+    opt = cli.parse(`commit-and-tag-version ${opt}`);
+  }
+  return require('../index')(opt);
+}
+
+function writePackageJson(version, option) {
+  const pkg = Object.assign({}, option, { version });
+  fs.writeFileSync('package.json', JSON.stringify(pkg), 'utf-8');
+}
+
+function writeHook(hookName, causeError, script) {
+  shell.mkdir('-p', 'scripts');
+  let content = script || 'console.error("' + hookName + ' ran")';
+  content += causeError ? '\nthrow new Error("' + hookName + '-failure")' : '';
+  fs.writeFileSync('scripts/' + hookName + '.js', content, 'utf-8');
+  fs.chmodSync('scripts/' + hookName + '.js', '755');
+}
+
+function setup() {
+  shell.rm('-rf', 'test-temp');
+  shell.config.silent = true;
+  shell.mkdir('test-temp');
+  shell.cd('test-temp');
+  shell.exec('git init');
+  shell.exec('git config commit.gpgSign false');
+  shell.exec('git config core.autocrlf false');
+  shell.exec('git commit --allow-empty -m"root-commit"');
+  writePackageJson('1.0.0');
+}
+
+function reset() {
+  shell.cd('../');
+  shell.rm('-rf', 'test-temp');
+
+  // Clear mock calls but don't reset implementation
```

Comment:
Optional: You could rename this function `clear` and then you wouldn't need this comment (probably best split into `resetShell()` and `clearMocks()`). It's only a minor improvement though.

Reply:
Done, did a bit of tidying of other tests and set the "Temp" dirs to have a consistent naming scheme in the integration tests, but they all now use different directories, which should add a backstop to help reduce concurrency issues if they end up getting run in parallel